pypeclub/OpenPype
# Workfiles tool: Save as published workfiles

## Brief description
Give option to save as published workfiles without storing it in temp folder.

## Description
Removed `View` and added 2 new buttons into published workfiles context.
1. `Copy & Open` which will save published workfile to it's workarea
2. `Choose different context` which will allow to change context of workarea where the file will be stored
When button n2 is pressed artist can change context of asset and task and confirm it with `Copy & Open` or cancel the operation with `Cancel`. During this time he can see other workfiles in workarea but the view is disabled.

## Changes
- removed `View` button from workfiles tool
- added `Copy & Open` and `Choose different context` buttons into published workfiles
- Save as dialog does not allow to change extension when published workfile is opened
- the `Published` checkbox is hidden if saving is disabled

## Screenshots
### When **Published** is checked
![image](https://user-images.githubusercontent.com/43494761/161056543-92a01c83-94d6-4b5c-9b41-a4b1cd0dd78c.png)
### When **Choose different context** is clicked
![image](https://user-images.githubusercontent.com/43494761/161059690-2f249267-9de8-4477-b547-834d4281aeb9.png)

## Testing notes:
1. Open workfiles tool in host
2. Check published
3. Try both `Copy & Open` and `Choose different context` buttons and play with it
